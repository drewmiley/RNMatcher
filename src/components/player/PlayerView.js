import React, { Component } from 'react';
import {
    Text,
    TouchableHighlight,
    View
} from 'react-native';

export default class PlayerView extends Component {
    constructor(props) {
        super(props);
        this.shouldComponentUpdate = false;
    };
    render() {
        return <View>
            <Text>
                {this.props.name} - {this.props.rating}
            </Text>
            <TouchableHighlight onPress={() => this.props.match(this.props.players, this.props.id)}>
                <Text>Match</Text>
            </TouchableHighlight>
        </View>
    };
};