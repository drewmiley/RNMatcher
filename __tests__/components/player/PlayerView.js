import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import PlayerView from '../../../src/components/player/PlayerView';

describe('PlayerView', () => {

    describe('Rendering', () => {

        it('should match the snapshot', () => {
            const component = renderer.create(
                <PlayerView />
            );            
            expect(component).toMatchSnapshot();
        });

    });

});