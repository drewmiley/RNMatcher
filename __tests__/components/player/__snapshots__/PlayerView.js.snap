exports[`PlayerView Rendering should match the snapshot 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail">
     - 
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail">
      Match
    </Text>
  </View>
</View>
`;
